src/components/pages/Home/Onboarding/Stepper.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { ReactElement } from 'react'
+import styles from './Stepper.module.css'
+import classNames from 'classnames/bind'
+import Button from '../../../atoms/Button'
+import { NavigationDirections } from '.'
+import { useWeb3 } from '../../../../providers/Web3'
+import { toast } from 'react-toastify'
+import { getErrorMessage } from '../../../../utils/onboarding'
+
+const cx = classNames.bind(styles)
+
+export default function Stepper({
+  stepLabels,
+  currentStep,
+  onboardingCompleted,
+  setCurrentStep,
+  setNavigationDirection
+}: {
+  stepLabels: string[]
+  currentStep: number
+  onboardingCompleted: boolean
+  setCurrentStep: (step: number) => void
+  setNavigationDirection: (direction: NavigationDirections) => void
+}): ReactElement {
+  const { accountId, balance, networkId, web3Provider } = useWeb3()
+
+  const handleClick = (newStep: number) => {
+    if (newStep === stepLabels.length - 1 && !onboardingCompleted) {
+      toast.error(
+        getErrorMessage({
+          accountId,
+          web3Provider: !!web3Provider,
+          networkId,
+          balance
+        })
+      )
+      return
+    }
+
+    currentStep > newStep
+      ? setNavigationDirection(NavigationDirections.PREV)
+      : setNavigationDirection(NavigationDirections.NEXT)
```

Comment:
nitpick(blocking): This piece of logic should be not handled by the clicker but by a dedicated function. Abstraction. Try to keep the logic of the click handler to a high level. It sets the navigation direction and sets the current step. Currently, you are also calculating which step to set. This should be none of your business.